lang/cs/Org.Apache.REEF.IO/PartitionedData/FileSystem/FileSystemInputPartition.cs

```diff
@@ -85,55 +134,14 @@ private void Initialize()
         /// <returns></returns>
         public T GetPartitionHandle()
         {
-            if (_copyToLocal)
-            {
-                if (!_isInitialized)
-                {
-                    Initialize();
-                }
-
-                return _fileSerializer.Deserialize(_localFiles);
-            }
-            return _fileSerializer.Deserialize(_filePaths);
-        }
-
-        private void CopyFromRemote()
-        {
-            string localFileFolder = _tempFileCreator.CreateTempDirectory("-partition-");
-            Logger.Log(Level.Info, string.Format(CultureInfo.CurrentCulture, "Local file temp folder: {0}", localFileFolder));
-
-            foreach (var sourceFilePath in _filePaths)
+            lock (_lock)
             {
-                Uri sourceUri = _fileSystem.CreateUriForPath(sourceFilePath);
-                Logger.Log(Level.Info, string.Format(CultureInfo.CurrentCulture, "sourceUri {0}: ", sourceUri));
-                if (!_fileSystem.Exists(sourceUri))
+                if (!_localFiles.IsPresent())
```

Comment:
Downloading files is also a form of caching, not all caching needs to go into memory.

What I am proposing in REEF-1357 is this:
1. There will only be one implementation of `IInputPartition`, `DefaultInputPartition`, and we will deprecate `FileSystemInputPartition`.
2. `DefaultIntputPartition` will have fields `IDataLoader` and `ISet<IDataCache>`, where `IDataLoader` downloads the data from, perhaps, a remote source, and `ISet<IDataCache>` sets the caching policy (Remote vs Disk vs Mem) based on the implementation that is bound.
3. Essentially REEF-1357 aims to decouple data loading with data caching. The new FileSystem file downloading functions will be moved into a `FileSystemDataLoader : IDataLoader`.